datadog_checks_base/datadog_checks/base/utils/db/core.py

```diff
@@ -94,83 +99,83 @@ def compile_queries(self):
 
     def execute(self, extra_tags=None):
         """This method is what you call every check run."""
-        logger = self.check.log
+        global_tags = list(self.tags)
         if extra_tags:
-            global_tags = list(extra_tags)
-            global_tags.extend(self.tags)
-        else:
-            global_tags = self.tags
+            global_tags.extend(list(extra_tags))
 
         for query in self.queries:
             query_name = query.name
-            query_columns = query.columns
-            query_extras = query.extras
-            query_tags = query.tags
-            num_columns = len(query_columns)
+            query_columns = query.column_transformers
+            extra_transformers = query.extra_transformers
+            query_tags = query.custom_tags
 
             try:
                 rows = self.execute_query(query.query)
             except Exception as e:
                 if self.error_handler:
-                    logger.error('Error querying %s: %s', query_name, self.error_handler(str(e)))
+                    self.logger.error('Error querying %s: %s', query_name, self.error_handler(str(e)))
                 else:
-                    logger.error('Error querying %s: %s', query_name, e)
+                    self.logger.error('Error querying %s: %s', query_name, e)
 
                 continue
 
             for row in rows:
-                if not row:
-                    logger.debug('Query %s returned an empty result', query_name)
+                if not self._is_row_valid(query, row):
                     continue
 
-                if num_columns != len(row):
-                    logger.error(
-                        'Query %s expected %d column%s, got %d',
-                        query_name,
-                        num_columns,
-                        's' if num_columns > 1 else '',
-                        len(row),
-                    )
-                    continue
+                query_result = {}
```

Comment:
We should keep the name `sources` to match docs https://datadoghq.dev/integrations-core/base/databases/#datadog_checks.base.utils.db.transform.ExtraTransformers